plans/003-snapshot-leaf-context-neighbors.md

```
@@ -0,0 +1,278 @@
+  internal nodes invisible to callback code.
+- `Array::copy` / `ArrayView` — checked but rejected. A copied array adds O(b)
+  allocation per operation, while a view still aliases the mutable array and
+  exposes `BTreeNode` values containing nested mutable arrays.
+- `Option` — reuse for captured absent/present neighbors.
+
+## Commands you will need
+
+| Purpose | Command | Expected on success |
+|---|---|---|
+| API/reference preflight | `NEW_MOON_MOD=0 moon ide doc 'Array::get'; NEW_MOON_MOD=0 moon ide find-references LeafContext` | `Array::get` is available; references are listed, or IDE failure is recorded and grep fallback used |
+| Grep fallback | `rg -n --hidden --no-ignore --glob '!**/_build/**' --glob '!**/.mooncakes/**' 'LeafContext|\.children' lib/btree order-tree event-graph-walker` | no production `ctx.children` consumer outside `lib/btree` |
```

markdownlint-cli2 (0.23.1)

[warning] Table column count
Expected: 3; Actual: 4; Too many cells, extra data will be missing

(MD056, table-column-count)